journal.py

```diff
@@ -243,7 +264,7 @@ def main():
     config.add_route('login', '/login')
     config.add_route('logout', '/logout')
     config.add_route('detail', '/detail/{id:\d+}')
-    config.add_route('edit', '/edit/{id:\d+}')
+    config.add_route('edit', '/edit')
```

Comment:
I'm curious why you removed the ID from the url, and changed it to be a query parameter?